whois-api/src/test/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParserTest.java

```diff
@@ -27,23 +28,32 @@ public void parse_permanent_delivery_failure_message_rfc822() throws Exception {
         final MessageInfo bouncedMessage = subject.parse(MimeMessageProvider.getUpdateMessage("permanentFailureMessageRfc822.mail"));
 
         assertThat(bouncedMessage.messageId(), is("XXXXXXXX-5AE3-4C58-8E3F-860327BA955D@ripe.net"));
-        assertThat(bouncedMessage.emailAddress(), is("nonexistant@host.org"));
+        assertThat(bouncedMessage.emailAddresses().get(0), is("nonexistant@host.org"));
+    }
+
+    @Test
+    public void parse_permanent_delivery_failure_multiple_recipients_rfc822() throws Exception {
+        final MessageInfo bouncedMessage = subject.parse(MimeMessageProvider.getUpdateMessage("permanentFailureMessageRfc822MultipleRecipients.mail"));
+
+        assertThat(bouncedMessage.messageId(), is("XXXXXXXX-5AE3-4C58-8E3F-860327BA955D@ripe.net"));
+        assertThat(bouncedMessage.emailAddresses(), containsInAnyOrder("nonexistant@host.org", "nonexistant1@host.org"));
```

Comment:
How are *both* addresses bounced? I only see in `permanentFailureMessageRfc822MultipleRecipients.mail` that `nonexistant@host.org` bounced, not `nonexistant1@host.org`

Reply:
both appear in the rfc822
To: nonexistant@host.org, nonexistant1@host.org

probably this use case is incomplete, we need a failure per each

Reply:
ok, added